From bbad53d3d2d3c5c0d82df5357bbc907590f52af3 Mon Sep 17 00:00:00 2001
From: Patrick Spendrin <ps_ml@gmx.de>
Date: Mon, 28 Apr 2014 16:40:44 +0200
Subject: [PATCH 17/28] disable extended attributes support on windows

At the moment, there is no support available for extended attributes for Windows/NTFS. So we simply disable it.
---
 src/file/lib/baloo_xattr_p.h   | 6 ++++++
 src/file/lib/xattrdetector.cpp | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/src/file/lib/baloo_xattr_p.h b/src/file/lib/baloo_xattr_p.h
index fefc3d9..de7bb42 100644
--- a/src/file/lib/baloo_xattr_p.h
+++ b/src/file/lib/baloo_xattr_p.h
@@ -51,6 +51,8 @@ inline ssize_t baloo_getxattr(const QString& path, const QString& name, QString*
     const ssize_t size = getxattr(encodedPath, attributeName, NULL, 0, 0, 0);
 #elif defined(Q_OS_FREEBSD) || defined(Q_OS_NETBSD)
     const ssize_t size = extattr_get_file(encodedPath, EXTATTR_NAMESPACE_USER, attributeName, NULL, 0);
+#elif defined(Q_OS_WIN)
+    const ssize_t size = 0;
 #endif
 
     if (size <= 0) {
@@ -68,6 +70,8 @@ inline ssize_t baloo_getxattr(const QString& path, const QString& name, QString*
     const ssize_t r = getxattr(encodedPath, attributeName, data.data(), size, 0, 0);
 #elif defined(Q_OS_FREEBSD) || defined(Q_OS_NETBSD)
     const ssize_t r = extattr_get_file(encodedPath, EXTATTR_NAMESPACE_USER, attributeName, data.data(), size);
+#elif defined(Q_OS_WIN)
+    const ssize_t r = 0;
 #endif
 
     *value = QString::fromUtf8(data);
@@ -94,6 +98,8 @@ inline int baloo_setxattr(const QString& path, const QString& name, const QStrin
 #elif defined(Q_OS_FREEBSD) || defined(Q_OS_NETBSD)
     const ssize_t count = extattr_set_file(encodedPath, EXTATTR_NAMESPACE_USER, attributeName, attributeValue, valueSize);
     return count == -1 ? -1 : 0;
+#elif defined(Q_OS_WIN)
+    return -1;
 #endif
 }
 
diff --git a/src/file/lib/xattrdetector.cpp b/src/file/lib/xattrdetector.cpp
index 9bff838..4bc5d96 100644
--- a/src/file/lib/xattrdetector.cpp
+++ b/src/file/lib/xattrdetector.cpp
@@ -101,6 +101,9 @@ XattrDetector::~XattrDetector()
 
 bool XattrDetector::isSupported(const QString& path)
 {
+#ifdef Q_OS_WIN
+    return false;
+#endif
     if (!d->m_initialized)
         d->init();
 
-- 
1.9.0

